rpm/0002-Implement-Sailfish-OS-specific-privileged-data-optio.patch

```diff
@@ -180,90 +176,149 @@ index 7a0d5283..d42791fc 100644
 +			id_cnt = 0;
 +			return map_data;
 +		}
-+
+ 
+-		char *map;
+-		uid_t uid = getuid();
+-		if (asprintf(&map, "%d %d 1", uid, uid) == -1)
 +		// UIDs
 +		if (asprintf(&map_path, "/proc/%d/uid_map", child) == -1)
- 	 		errExit("asprintf");
-- 		EUID_ROOT();
--	 	update_map(map, map_path);
+ 			errExit("asprintf");
 +		id_add(0);
 +		id_add(euid_data.uid);
 +		id_add(euid_data.privileged_uid);
-+		EUID_ROOT();
+ 		EUID_ROOT();
+-		update_map(map, map_path);
 +		update_map(id_map(), map_path);
- 	 	EUID_USER();
--	 	free(map);
- 	 	free(map_path);
+ 		EUID_USER();
+-		free(map);
+ 		free(map_path);
  
--	 	// gid file
+-		// gid file
 +		// GIDs
  		if (asprintf(&map_path, "/proc/%d/gid_map", child) == -1)
  			errExit("asprintf");
--	 	char gidmap[1024];
--	 	char *ptr = gidmap;
--	 	*ptr = '\0';
+-		char gidmap[1024];
+-		char *ptr = gidmap;
+-		*ptr = '\0';
 -
--	 	// add user group
--	 	gid_t gid = getgid();
--	 	sprintf(ptr, "%d %d 1\n", gid, gid);
--	 	ptr += strlen(ptr);
+-		// add user group
+-		gid_t gid = getgid();
+-		sprintf(ptr, "%d %d 1\n", gid, gid);
+-		ptr += strlen(ptr);
  
+-		gid_t g;
 +		id_add(0);
 +		id_add(euid_data.gid);
 +		id_add(euid_data.primary_gid);
 +		id_add(euid_data.privileged_gid);
- 	 	if (!arg_nogroups) {
- 		 	//  add firejail group
--		 	gid_t g = get_group_id("firejail");
--		 	if (g) {
--		 		sprintf(ptr, "%d %d 1\n", g, g);
--		 		ptr += strlen(ptr);
--		 	}
+ 		if (!arg_nogroups || !check_can_drop_all_groups()) {
+ 			// add audio group
+ 			if (!arg_nosound) {
+-				g = get_group_id("audio");
+-				if (g) {
+-					sprintf(ptr, "%d %d 1\n", g, g);
+-					ptr += strlen(ptr);
+-				}
++				id_add(get_group_id("audio"));
+ 			}
 -
-+			id_add(get_group_id("firejail"));
- 		 	//  add tty group
--		 	g = get_group_id("tty");
--		 	if (g) {
--		 		sprintf(ptr, "%d %d 1\n", g, g);
--		 		ptr += strlen(ptr);
--		 	}
+ 			// add video group
+ 			if (!arg_novideo) {
+-				g = get_group_id("video");
+-				if (g) {
+-					sprintf(ptr, "%d %d 1\n", g, g);
+-					ptr += strlen(ptr);
+-				}
++				id_add(get_group_id("video"));
+ 			}
 -
-+			id_add(get_group_id("tty"));
- 		 	//  add audio group
--		 	g = get_group_id("audio");
--		 	if (g) {
--		 		sprintf(ptr, "%d %d 1\n", g, g);
--		 		ptr += strlen(ptr);
--		 	}
+-			// add render/vglusers group
+-			if (!arg_no3d) {
+-				g = get_group_id("render");
```

Comment:
vglusers is for VirtualGl which is not used on Sailfish OS. Lp group is used in cups to allow printing.

Partially these groups might be Debian specific such as the render group:
https://wiki.debian.org/SystemGroups

Note that these groups are required if logind doesn't manage device access based on the udev "uaccess" , it makes sense in case we add them that we should mark this as a FIXME and create a bug for it.

Reply:
That sounds reasonable. I'd be fine with FIXME and a task.